modules/data-entry/src/main/java/io/uhndata/cards/dataentry/internal/serialize/FlattenFormProcessor.java

```diff
@@ -0,0 +1,128 @@
+ * under the License.
+ */
+package io.uhndata.cards.dataentry.internal.serialize;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import javax.jcr.Node;
+import javax.jcr.NodeIterator;
+import javax.jcr.RepositoryException;
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+import javax.json.JsonValue;
+
+import org.apache.sling.api.resource.Resource;
+import org.osgi.service.component.annotations.Component;
+
+import io.uhndata.cards.serialize.spi.ResourceJsonProcessor;
+
+@Component(immediate = true)
+public class FlattenFormProcessor implements ResourceJsonProcessor
+{
+    private static final int ID_LENGTH = 36;
+
+    private ThreadLocal<Map<String, JsonObject>> childrenJsons = ThreadLocal.withInitial(HashMap::new);
+    private ThreadLocal<JsonObjectBuilder> jsonSection = ThreadLocal.withInitial(Json::createObjectBuilder);
+    private ThreadLocal<Map<String, JsonObjectBuilder>> jsonSectionMap = ThreadLocal.withInitial(HashMap::new);
+
+    @Override
+    public String getName()
+    {
+        return "flatten";
+    }
+
+    @Override
+    public int getPriority()
+    {
+        return 100;
+    }
+
+    @Override
+    public boolean canProcess(Resource resource)
+    {
+        return resource.isResourceType("cards/Form");
+    }
+
+    @Override
+    public JsonValue processChild(Node node, Node child, JsonValue input, Function<Node, JsonValue> serializeNode)
+    {
+        if (input == null) {
+            return null;
+        }
+        try {
+            if (child.isNodeType("cards:Answer")) {
+                this.childrenJsons.get().put(child.getIdentifier(), input.asJsonObject());
+                return null;
+            } else if (child.isNodeType("cards:AnswerSection")) {
+                return null;
+            }
+        } catch (RepositoryException e) {
+            // Shouldn't happen
+        }
+        return input;
+    }
+
+    @Override
+    public void leave(Node node, JsonObjectBuilder json, Function<Node, JsonValue> serializeNode)
+    {
+        addAnswers(node, json);
+    }
+
+    @Override
+    public void end(Resource resource)
+    {
+        this.childrenJsons.remove();
+        this.jsonSection.remove();
+        this.jsonSectionMap.remove();
+    }
+
+    private void addAnswers(final Node node, final JsonObjectBuilder json)
+    {
+        try {
+            // The first time we encounter a section, we create a JsonArrayBuilder for it,
+            // and add its children in the output.
+            // While leaving the section we add all its children saved in Map getting them by section id.
+            final NodeIterator children = node.getNodes();
+            while (children.hasNext()) {
+                final Node child = children.nextNode();
+                final String childId = child.getIdentifier();
+                final String idWithoutParent = childId.substring(childId.length() - ID_LENGTH);
```

Comment:
It's not a good practice to assume a certain length for identifiers. Instead, we can rely on the known structure of the string and pull out the last segment:
```suggestion
                final String idWithoutParent = childId.substring(childId.lastIndexOf("/") + 1);
```